recipes-openxt/tss/tpm2-tools/tpm2-tools-lib-support.patch

```diff
@@ -270,111 +321,123 @@ index 4a998d2..caf976f 100644
 +#define TEST_PCR_SELECT_BIT( pcrSelection, pcr ) \
 +    ((pcrSelection).pcrSelect[( (pcr)/8 )] & ( 1 << ( (pcr) % 8) ))
 +
- 
  #endif /* SRC_PCR_H_ */
+diff --git a/lib/tpm2_util.c b/lib/tpm2_util.c
+index 57d6c76..ca1230a 100644
+--- a/lib/tpm2_util.c
++++ b/lib/tpm2_util.c
+@@ -106,6 +106,18 @@ bool tpm2_util_string_to_uint32(const char *str, uint32_t *value) {
+     return true;
+ }
+ 
++int tpm2_util_string_to_byte_structure(const char *inStr, UINT16 *byteLength, BYTE *byteBuffer) {
++    if(inStr == NULL || byteLength == NULL || byteBuffer == NULL)
++        return -1;
++    if(*byteLength <= strlen(inStr))
++        return -2;
++
++    *byteLength = strlen(inStr);
++    memcpy(byteBuffer, inStr, *byteLength);
++    byteBuffer[*byteLength] = '\0';
++    return 0;
++}
++
+ int tpm2_util_hex_to_byte_structure(const char *inStr, UINT16 *byteLength,
+         BYTE *byteBuffer) {
+     int strLength; //if the inStr likes "1a2b...", no prefix "0x"
+diff --git a/lib/tpm2_util.h b/lib/tpm2_util.h
+index e803dc1..3053350 100644
+--- a/lib/tpm2_util.h
++++ b/lib/tpm2_util.h
+@@ -108,8 +108,8 @@ struct TPM2B {
+         } while ((__result & 0x0000ffff) == TPM2_RC_RETRY); \
+         __result;                                          \
+     })
+-
+-int tpm2_util_hex_to_byte_structure(const char *inStr, UINT16 *byteLenth, BYTE *byteBuffer);
++int tpm2_util_string_to_byte_structure(const char *inStr, UINT16 *byteLength, BYTE *byteBuffer);
++int tpm2_util_hex_to_byte_structure(const char *inStr, UINT16 *byteLength, BYTE *byteBuffer);
+ 
+ /**
+  * Appends a TPM2B_DIGEST buffer to a TPM2B_MAX buffer.
 diff --git a/lib/tpm_hash.c b/lib/tpm_hash.c
-index fcd107b..a28afbb 100644
+index aa34741..3706d7b 100644
```

Comment:
This patch should be refreshed with the recommended configuration:
https://openxt.atlassian.net/wiki/spaces/DC/pages/20905986/Dealing+with+quilt

Reply:
Addressed.